src/typemap.c

```diff
@@ -280,16 +375,24 @@ static int jl_typemap_intersection_array_visitor(jl_array_t *a, jl_value_t *ty,
         JL_GC_PROMISE_ROOTED(t);
         if (t == jl_nothing || t == NULL)
             continue;
-        // `t` is a leaftype, so intersection test becomes subtype
-        if (ty == (jl_value_t*)jl_any_type || // easy case: Any always matches
-            (tparam
-             ? (jl_typeof(t) == ty || jl_isa(t, ty)) // (Type{t} <: ty), where is_leaf_type(t) => isa(t, ty)
-             : (t == ty || jl_subtype(t, ty)))) {
+        if (tparam & 2) {
```

Comment:
Could use a comment on this function saying what the bits of `tparam` mean.

Reply:
Whether it's the TypeName and whether it's the parameter of Type{} (but agree it should have a comment)